tiledb/ml/models/base.py

```diff
@@ -73,38 +104,37 @@ def preview(self, **kwargs):
-    ) -> dict:
+    def set_file_properties(self):
         """
-        Abstract method that returns model array's file properties.
-        :param framework: str. Machine learning framework.
-        :param framework_version: str. Machine learning framework's version.
-        :param preview: str. Machine learning model's preview.
-        :return: dict. Dictionary with model's file properties
+        Method that sets model array's file properties.
         """
-        return {
-            self.file_properties_obj.ML_FRAMEWORK.value: framework,
-            self.file_properties_obj.STAGE.value: "STAGING",
-            self.file_properties_obj.PYTHON_VERSION.value: platform.python_version(),
-            self.file_properties_obj.FRAMEWORK_VERSION.value: framework_version,
-            self.file_properties_obj.MODEL_PREVIEW.value: preview,
+        self._file_properties = {
+            ModelFileProperties.ML_FRAMEWORK.value: self.framework,
+            ModelFileProperties.ML_FRAMEWORK_VERSION.value: self.framework_version,
+            ModelFileProperties.STAGE.value: "STAGING",
+            ModelFileProperties.PYTHON_VERSION.value: platform.python_version(),
+            ModelFileProperties.PREVIEW.value: self.preview(),
         }
 
-    @staticmethod
-    def update_model_metadata(array: tiledb.Array, meta: dict):
-        for key, value in meta.items():
-            array.meta[key] = json.dumps(value).encode("utf8")
-
-
-@unique
-class ModelFileProperties(Enum):
-    """
-    Enum Class that contains all model array file properties.
-    """
+    def update_model_metadata(self, array: tiledb.Array, meta: dict):
+        """
+        This method updates the metadata in a TileDB model array. File properties also go in the metadata section.
+        :param array: tiledb.Array. A TileDB model array.
+        :param meta: dict. A dictionary with the <key, value> pairs that will be inserted in array's metadata.
+        """
+        # Raise ValueError in case users provide metadata with the same keys as file properties.
+        if meta:
+            if meta.keys() & self._file_properties.keys():
+                raise ValueError(
+                    "Please avoid using file property key names as metadata keys!"
+                )
+            else:
+                for key, value in {**meta, **self._file_properties}.items():
+                    array.meta[key] = json.dumps(value).encode("utf8")
+        else:
+            # In case meta is None, insert only file properties in the metadata.
+            for key, value in self._file_properties.items():
+                array.meta[key] = json.dumps(value).encode("utf8")
 
-    ML_FRAMEWORK = "ML_FRAMEWORK"
-    FRAMEWORK_VERSION = "ML_FRAMEWORK_VERSION"
-    STAGE = "STAGE"
-    PYTHON_VERSION = "PYTHON_VERSION"
-    MODEL_PREVIEW = "PREVIEW"
+    def set_cloud_uri(self, s3_prefix: str, uri: str) -> str:
+        return "tiledb://{}/{}".format(self.namespace, os.path.join(s3_prefix, uri))
```

Comment:
Can be inlined, called only once

Reply:
Cannot be mocked though.

Reply:
Why does it need mocks for both this and `get_s3_prefix`? I think you can move the `get_s3_prefix` call inside `set_cloud_uri` and have only function to mock.